### PR TITLE
Kyber pricefeed

### PR DESCRIPTION
**Pull request by schmidsi**
Tests as example how to use Kyber as PriceFeed.

### CODE REVIEW

File: src/contracts/prices/KyberPriceFeed.sol [hunk 1]
```diff
@@ -29,33 +29,11 @@ contract KyberPriceFeed is PriceSourceInterface, DSThing {
...
-        bytes32 quoteAssetName,
-        bytes8 quoteAssetSymbol,
-        uint quoteAssetDecimals,
-        string quoteAssetUrl,
-        string quoteAssetIpfsHash,
-        address[2] quoteAssetBreakInBreakOut,
-        uint[] quoteAssetStandards,
-        bytes4[] quoteAssetFunctionSignatures,
-        address ofGovernance
+        address ofQuoteAsset
```
Comment (GauthamN) [inline src/contracts/prices/KyberPriceFeed.sol]: Have to pass in registry with new changes. I will make the changes

File: src/contracts/exchanges/thirdparty/kyber/transactions/setBaseRate.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+  const baseSell = isBuySell(prices)
+    ? prices.map(p => toAtomic(p.sell))
+    : prices.map(p => toAtomic(p));
+
+  const blockNumber =
+    givenBlockNumber || (await getLatestBlock(environment)).number;
+
+  const zeroExArray = isBuySell(prices)
+    ? prices.map(p => '0x')
```
Comment (GauthamN) [inline src/contracts/exchanges/thirdparty/kyber/transactions/setBaseRate.ts]: zeroExArray is constructed by pushing 0's to an array (upto 14) and converting that array to hex. If more than 14 tokens, additional 0's are added to another array and converted. So overall zeroExArray would look like [hex1, hex2..n] where n = no of tokens % 14

File: src/contracts/exchanges/thirdparty/kyber/transactions/setBaseRate.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    : prices.map(p => '0x');
+
+  const zeroArray = isBuySell(prices) ? prices.map(p => 0) : prices.map(p => 0);
```
[the remark this answers is not in the document]
Comment (schmidsi) [reply src/contracts/exchanges/thirdparty/kyber/transactions/setBaseRate.ts]: > Does this mean zeroArray is going to be an array like [0, 0, 0.. n] where n is number of tokens.
Yes

> It's supposed to be [0, 0 .. n] where n = no of tokens % 14
Ok. Actually, I did not implement the whole logic yet, it was too complicated. For our testing-use-case, it is actually sufficient to just update the price of 1 or two tokens. On kovan we do not need to update prices by ourselves. Although @fubhy is probably interested in these functions (?)

[the remark this answers is not in the document]
Comment (jennazenk) [reply src/contracts/exchanges/thirdparty/kyber/transactions/setBaseRate.ts]: yes you're right @schmidsi these functions will be interesting for @fubhy and reserve woorton. But we also need them to update the prices of our reserve on kovan (unless we keep the current reserve we have running on Kovan?)

[the remark this answers is not in the document]
Comment (schmidsi) [reply src/contracts/exchanges/thirdparty/kyber/transactions/setBaseRate.ts]: But for just read the price of existing Kovan reserves, this PR is good to merge in my opinion.

[the remark this answers is not in the document]
Comment (jennazenk) [reply src/contracts/exchanges/thirdparty/kyber/transactions/setBaseRate.ts]: ok sounds good!

**Pull request merged**